test/scheduler_test.rb

```diff
@@ -456,6 +456,52 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
+  test 'gracefully shuts down rufus-scheduler threads' do
+    class BeforeEnqueueJob
+      @queue = :quick
+
+      def self.before_enqueue_example(*)
+        if Resque.redis.setnx('before_enqueue_job:enqueued', 'true')
```

Comment:
hmm I think the intent here could be a bit more clear to make the test easier to understand. How about something like this:
```ruby
class << self
  def before_enqueue_example(*)
    job_has_already_been_enqueued = set_before_enqueue_flag(true)
    return false if job_has_already_been_enqueued
    sleep 5
    true
  end

  private

  def set_before_enqueue_flag(value)
    Resque.redis.setnx('before_enqueue_job:enqueued', value)
  end
end
```